crates/core/tedge_actors/src/builders.rs

```diff
@@ -69,6 +70,12 @@ pub trait MessageSource<M: Message, Config> {
     fn register_peer(&mut self, config: Config, sender: DynSender<M>);
 }
 
+/// The builder of a MessageBox must implement this trait to receive requests from the runtime
+pub trait RuntimeRequestSink {
```

Comment:
Why did we need this specialised  `RuntimeRequestSink` which effectively is just a `MessageSink<RuntimeRequest>`? Why couldn't we just reuse `MessageSink<RuntimeRequest>`?

Reply:
This is unfortunate but required as using `MessageSink<RuntimeRequest>` leads to conflicting implementations. 

Indeed, `SimpleMessageBoxBuider<I,O>`  cannot implement both `MessageSink<I>` and `MessageSink<RuntimeRequest>` because these two implementations would conflict for  `SimpleMessageBoxBuider<RuntimeRequest,O>`. 
 
It would be nice if the rust compiler were deferring the error till a conflict actually arises rather than being only potential. For now, the simpler is to come with a specific trait.

```
conflicting implementations of trait `builders::MessageSink<runtime::RuntimeRequest>` for type `builders::SimpleMessageBoxBuilder<runtime::RuntimeRequest, _>`

    |
191 | impl<I: Message, O: Message> MessageSink<I> for SimpleMessageBoxBuilder<I, O> {
    | ----------------------------------------------------------------------------- first implementation here
...
197 | impl<I: Message, O: Message> MessageSink<RuntimeRequest> for SimpleMessageBoxBuilder<I, O> {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `builders::SimpleMessageBoxBuilder<runtime::RuntimeRequest, _>`
```